### PR TITLE
DRAFT Add AWS_MSK_IAM SASL Mechanism

### PR DESCRIPTION
**Pull request by cmaher**
Some comments are directly on the first commit: https://github.com/paxosglobal/kafka-go/commit/e6ed7e59525f499f5fddd8672297e6efbaa5afda

### CODE REVIEW

File: sasl/aws_msk_iam/msk_iam.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package aws_msk_iam
+
+import (
+	"context"
```
Comment (rahul-aravind-opti) [inline sasl/aws_msk_iam/msk_iam.go:4]: nit: if these are not formatted, lets format using `goimports`?

File: sasl/aws_msk_iam/msk_iam.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+func NewMechanism(brokerHost, awsRegion string, creds *credentials.Credentials) *Mechanism {
+	return &Mechanism{
+		BrokerHost: brokerHost,
+		AwsRegion:  awsRegion,
+		Signer:     sigv4.NewSigner(creds),
+		Expiry:     defaultSignExpiry,
+	}
+}
+
+func (m *Mechanism) Name() string {
```
Comment (rahul-aravind-opti) [inline sasl/aws_msk_iam/msk_iam.go:58]: Where is this being used?

File: sasl/aws_msk_iam/msk_iam_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		)
+	}
+
+	authMap := map[string]string{}
+	err = json.Unmarshal(auth, &authMap)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedMap := map[string]string{
```
Comment (rahul-aravind-opti) [inline sasl/aws_msk_iam/msk_iam_test.go:45]: can we simulate a test where the kafka broker rejects the authentication? we would need a mock mechanism that would return the failed auth response I guess. If its too much to add, feel free to ignore.

**Review by rahul-aravind-opti — COMMENTED** [review]
lgtm

File: sasl/aws_msk_iam/msk_iam_test.go (hunk 3 above)
Comment (cmaher) [reply sasl/aws_msk_iam/msk_iam_test.go:45]: The base package handles these tests for other sasl mechanisms, using real sasl options available on the actual kafka container.

File: sasl/aws_msk_iam/msk_iam.go (hunk 2 above)
Comment (cmaher) [reply sasl/aws_msk_iam/msk_iam.go:58]: this is called from the sasl-skeleton

**Pull request closed without merging**